### ISSUE TITLE
Increase aXe usage in Karma tests

### ISSUE THREAD

**Issue by swissspidy** [GoogleForCreators/web-stories-wp#7530]
## Task Description

Since #4778 we have a custom `toHaveNoViolations()` matcher in our Karma test suite to verify that a given node/tree doesn't have any obvious a11y violations.

However, it isn't actually being used at the moment since I never really announced it anywhere.

Now the question is how it can be best used in our code base to be of most use.

For comparison, in the e2e test suite we run a similar check for the whole page in an `afterEach` call.

Is that feasible for Karma as well?

Or should we just try to use this matcher manually in some places that we think it's useful in?

**Comment by swissspidy** [comment]
pinging @BrittanyIRL @barklund @choumx for thoughts

**Comment by BrittanyIRL** [comment]
Thoughts about trying it on new features that are getting tests written as we go? The first one that comes to mind is trying it out with the quick action menus tests that are getting written as we flesh the menus out. This would ideally make testing easier as axe should hopefully catch any poor a11y set up there.

**Comment by swissspidy** [comment]
Sounds like a good first step 👍

**Comment by BrittanyIRL** [comment]
We've got bandwidth in pea pod  - will take this and add more a11y checks throughout codebase. possibly multiple PRs depending on what needs fixing. Pointing big so we have padding to fix a11y errors found. anything that's massive can get it's own ticket :)

**Comment by BrittanyIRL** [comment]
Tests are added everywhere that makes sense in the editor, it should be easy to add as we go moving forward with new features.
